parser/src/moves.rs

```diff
@@ -153,17 +154,36 @@ pub(crate) fn spaces() -> RepeatedMove<PredicateMove<impl (for<'a> Fn(Token<'a>)
     repeat_n(1, space())
 }
 
-///A move to consume all spaces and escaped newlines
-pub(crate) fn word_sep() -> RepeatedMove<
-    OrMove<
+///A move to consume a space or escaped newline
+pub(crate) fn word_sep() -> OrMove<
+    PredicateMove<impl (for<'a> Fn(Token<'a>) -> bool) + Copy>,
+    AndThenMove<
+        PredicateMove<impl (for<'a> Fn(Token<'a>) -> bool) + Copy>,
         PredicateMove<impl (for<'a> Fn(Token<'a>) -> bool) + Copy>,
-        AndThenMove<
-            PredicateMove<impl (for<'a> Fn(Token<'a>) -> bool) + Copy>,
-            PredicateMove<impl (for<'a> Fn(Token<'a>) -> bool) + Copy>,
-        >,
     >,
 > {
-    repeat_n(1, space().or(of_type(BackSlash).and_then(of_type(NewLine))))
+    space().or(of_type(BackSlash).and_then(of_type(NewLine)))
```

Comment:
Is this change intentional? You are literally using `repeat_n(1,word_sep())` in the literal parser.